docs/writing_your_plugin.md

```diff
@@ -65,6 +65,14 @@ You can also replace any existing functionality this way.
     -   If you need custom concurrency (more complicated than a loop),
             triggered concurrency (by events fired in other plugins),
             or no concurrency then your plugin class should extend [`plugin`][13].
+			
+        - If you spin your own threads, they **must** wait for
+          `pb->lookup_impl<Stoplight>()->wait_for_ready()` the first time they
+          run. This allows the start of all threads in ILLIXR to be
+          synchronized.
```

Comment:
Should we enforce this with a new derived class for non-managed types?

Reply:
Non-managed threads are quite a special case, so I don't know if a solution would generalize well.